src/supervisor/watchers/handlers/index.ts

```diff
@@ -125,10 +136,105 @@ async function isSupportedWorkload(
   namespace: string,
   workloadKind: WorkloadKind,
 ): Promise<boolean> {
-  if (workloadKind !== WorkloadKind.DeploymentConfig) {
-    return true;
+  switch (workloadKind) {
+    case WorkloadKind.DeploymentConfig:
+      return await isDeploymentConfigSupported(namespace);
+    case WorkloadKind.CronJob:
+      return await isCronJobSupported(namespace);
+    case WorkloadKind.CronJobV1Beta1:
+      return await isCronJobV1Beta1Supported(namespace);
+    default:
+      return true;
+  }
+}
+
+async function isCronJobSupported(namespace: string): Promise<boolean> {
+  try {
+    const pretty = undefined;
+    const allowWatchBookmarks = undefined;
+    const continueToken = undefined;
+    const fieldSelector = undefined;
+    const labelSelector = undefined;
+    const limit = 1; // Try to grab only a single object
+    const resourceVersion = undefined; // List anything in the cluster
+    const resourceVersionMatch = undefined;
+    const timeoutSeconds = 10; // Don't block the snyk-monitor indefinitely
+    const attemptedApiCall =
+      await kubernetesApiWrappers.retryKubernetesApiRequest(() =>
+        k8sApi.batchClient.listNamespacedCronJob(
+          namespace,
+          pretty,
+          allowWatchBookmarks,
+          continueToken,
+          fieldSelector,
+          labelSelector,
+          limit,
+          resourceVersion,
+          resourceVersionMatch,
+          timeoutSeconds,
+        ),
+      );
+    return (
+      attemptedApiCall !== undefined &&
+      attemptedApiCall.response !== undefined &&
+      attemptedApiCall.response.statusCode !== undefined &&
+      attemptedApiCall.response.statusCode >= 200 &&
+      attemptedApiCall.response.statusCode < 300
+    );
+  } catch (error) {
+    logger.debug(
+      { error, workloadKind: WorkloadKind.CronJob },
+      'Failed on Kubernetes API call to list CronJob',
```

Comment:
should we have a bit more detail on this so we don't get pestered by support when they see this error like we do for DeploymentConfig?

Reply:
Both are now reduced to debug level, so they will show only if someone changes LOG_LEVEL=debug